aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -124,7 +126,10 @@ synchronized void startSubscriptions() throws DataStoreException {
 
         // Need to create a new buffer so we can properly handle retries and stop/start scenarios.
         // Re-using the same buffer has some unexpected results due to the replay aspect of the subject.
-        buffer = ReplaySubject.create();
+        // We allow mutations to reside for 10 minutes and collect a maximum of 100 for predicable
+        // memory usage.
+        buffer = ReplaySubject.createWithTimeAndSize(
+            DEFAULT_MAX_AGE_IN_MINUTES, TimeUnit.MINUTES, Schedulers.io(), DEFAULT_MAX_TOTAL_SUBSCRIPTIONS);
```

Comment:
@manueliglesias I'm comfortable with you just making a PR. It seems I have pointed out the problems and you all have a grasp on them now. Since you have already done the work, please go ahead and just merge in your change.